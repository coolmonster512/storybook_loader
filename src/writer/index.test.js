import mock from 'mock-fs';
import path from 'path';
import fs from 'fs';
import { writeFile, outputPath, templatePath } from './';

const template = fs.readFileSync(templatePath, 'utf-8');
const baseDir = path.resolve(__dirname, '../../');

beforeEach(() => {
  mock({
    [outputPath]: '',
    [templatePath]: template,
  });
});

afterEach(() => {
  mock.restore();
});

test('it should do what I think it should do', () => {
  const files = [
    path.resolve(__dirname, '../file1.js'),
    path.resolve(__dirname, '../sub/file2.js'),
    path.resolve(__dirname, '../../parent/file3.js'),
    path.resolve(__dirname, './sub/file4.js'),
    path.resolve(__dirname, './sub/sub/file5.js'),
  ];
  writeFile(baseDir, files);

  const contents = fs.readFileSync(outputPath, 'utf-8');

  expect(contents).toMatchSnapshot();
});
