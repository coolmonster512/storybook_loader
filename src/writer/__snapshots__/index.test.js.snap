// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`writeFile should perform expected work 1`] = `
"
// Auto-generated file created by react-native-storybook-loader
// Do not edit.
//
// https://github.com/elderfo/react-native-storybook-loader.git

function loadStories() {
  require('./parent/file3.js');
  require('./src/file1.js');
  require('./src/sub/file2.js');
  require('./src/writer/sub/file4.js');
  require('./src/writer/sub/sub/file5.js');
  
}

const stories = [
  './parent/file3.js',
  './src/file1.js',
  './src/sub/file2.js',
  './src/writer/sub/file4.js',
  './src/writer/sub/sub/file5.js',
  
];

module.exports = {
  loadStories,
  stories,
};
"
`;
